src/providers/on-chain-quote-provider.ts

```diff
@@ -282,30 +288,32 @@ export class OnChainQuoteProvider implements IOnChainQuoteProvider {
       baseBlockOffset: 0,
       rollback: { enabled: false },
     },
-    protected isMixedRouteQuoteProvider: boolean = false,
     protected quoterAddressOverride?: string
-  ) {
-    const quoterAddress = quoterAddressOverride
-      ? quoterAddressOverride
-      : isMixedRouteQuoteProvider
+  ) {}
+
+  private getQuoterAddress(isMixedRoutes: boolean): string {
+    if (this.quoterAddressOverride) {
+      return this.quoterAddressOverride;
+    }
+    const quoterAddress = isMixedRoutes
       ? MIXED_ROUTE_QUOTER_V1_ADDRESSES[this.chainId]
       : QUOTER_V2_ADDRESSES[this.chainId];
-
     if (!quoterAddress) {
       throw new Error(
-        `No address for the quoter contract on chain id: ${chainId}`
+        `No address for the quoter contract on chain id: ${this.chainId}`
       );
     }
-
-    this.quoterAddress = quoterAddress;
+    return quoterAddress;
   }
 
-  public async getQuotesManyExactIn(
+  public async getQuotesManyExactIn<
+    TRoute extends V3Route | V2Route | MixedRoute
+  >(
```

Comment:
I like using the generic type here because it lets us type the output, which helps a lot as we call this function to generate the corresponding `RouteWithValidQuote` classes which are distinct depending on the route type.